src/mono/wasm/debugger/DebuggerTestSuite/EvaluateOnCallFrameTests.cs

```diff
@@ -1147,8 +1147,61 @@ public async Task EvaluateLocalObjectFromAssemblyNotRelatedButLoaded()
                    );
             });
 
+        [ConditionalFact(nameof(RunningOnChrome))]
+        public async Task EvaluateNullObjectPropertiesPositive() => await CheckInspectLocalsAtBreakpointSite(
+            $"DebuggerTests.EvaluateNullableProperties", "Evaluate", 6, "Evaluate",
+            $"window.setTimeout(function() {{ invoke_static_method ('[debugger-test] DebuggerTests.EvaluateNullableProperties:Evaluate'); 1 }})",
+            wait_for_event_fn: async (pause_location) =>
+            {
+                var id = pause_location["callFrames"][0]["callFrameId"].Value<string>();
+
+                // we have no way of returning int? for null values, 
+                // so we return the last non-null class name
+                await EvaluateOnCallFrameAndCheck(id,
+                   ("list.Count", TNumber(1)),
+                   ("list!.Count", TNumber(1)),
+                   ("list?.Count", TNumber(1)),
+                   ("listNull", TObject("System.Collections.Generic.List<int>", is_null: true)),
+                   ("listNull?.Count", TObject("System.Collections.Generic.List<int>", is_null: true)),
+                   ("tc?.memberList?.Count", TNumber(2)),
+                   ("tc!.memberList?.Count", TNumber(2)),
+                   ("tc?.memberListNull?.Count", TObject("System.Collections.Generic.List<int>", is_null: true)),
+                   ("tc.memberListNull?.Count", TObject("System.Collections.Generic.List<int>", is_null: true)),
+                   ("tcNull?.memberListNull?.Count", TObject("DebuggerTests.EvaluateNullableProperties.TestClass", is_null: true)));
+            });
+
+        [ConditionalFact(nameof(RunningOnChrome))]
+        public async Task EvaluateNullObjectPropertiesNegative() => await CheckInspectLocalsAtBreakpointSite(
+            $"DebuggerTests.EvaluateNullableProperties", "Evaluate", 6, "Evaluate",
+            $"window.setTimeout(function() {{ invoke_static_method ('[debugger-test] DebuggerTests.EvaluateNullableProperties:Evaluate'); 1 }})",
+            wait_for_event_fn: async (pause_location) =>
+            {
+                var id = pause_location["callFrames"][0]["callFrameId"].Value<string>();
+                var (_, res) = await EvaluateOnCallFrame(id, "listNull.Count", expect_ok: false);
+                AssertEqual("Cannot access member \"Count\" of a null-valued object.",
+                    res.Error["result"]?["description"]?.Value<string>(), "wrong error message");
+                (_, res) = await EvaluateOnCallFrame(id, "listNull!.Count", expect_ok: false);
```

Comment:
And `// await CheckEvaluateFail("x?.ToString()", "Operation '?' not allowed on type 'int'");` can be a case for a future PR.

Reply:
https://github.com/dotnet/runtime/issues/70337